truss/templates/server/common/truss_server.py

```diff
@@ -1,248 +1,169 @@
-import json  # noqa: E402
-import logging  # noqa: E402
-import os
-import sys
-from http import HTTPStatus  # noqa: E402
-from threading import Thread
-from typing import List
-
-import numpy as np  # noqa: E402
-import tornado.web  # noqa: E402
-from common.lib_support import ensure_kfserving_installed
-from common.serialization import DeepNumpyEncoder  # noqa: E402
-from common.serialization import (  # noqa: E402
+import json
+import logging
+from typing import Dict, Optional, Union
+
+import kserve
+import kserve.errors as errors
+import numpy as np
+from common.serialization import (
+    DeepNumpyEncoder,
     truss_msgpack_deserialize,
     truss_msgpack_serialize,
 )
-from common.util import (  # noqa: E402
-    assign_request_to_inputs_instances_after_validation,
-)
-from kfserving import KFModel
-from kfserving.handlers.http import HTTPHandler  # noqa: E402
-from kfserving.kfserver import HealthHandler, KFServer, ListHandler, LivenessHandler
+from common.util import assign_request_to_inputs_instances_after_validation
+from fastapi import Depends, FastAPI, Request
+from fastapi.responses import ORJSONResponse
+from fastapi.routing import APIRoute as FastAPIRoute
+from kserve.errors import InvalidInput
+from kserve.handlers import DataPlane, ModelRepositoryExtension, V1Endpoints
 from model_wrapper import ModelWrapper
-from pythonjsonlogger import jsonlogger  # noqa: E402
-from tornado.httpserver import HTTPServer
-from tornado.ioloop import IOLoop
-
-ensure_kfserving_installed()
+from starlette.responses import Response
 
 logger = logging.getLogger(__name__)
 
 
-def _configure_logging():
-    json_log_handler = logging.StreamHandler()
-    json_log_handler.setFormatter(
-        jsonlogger.JsonFormatter("%(asctime)s %(levelname)s %(message)s")
-    )
-    logger = logging.getLogger()
-    for handler in logger.handlers:
-        logger.removeHandler(handler)
-    logger.addHandler(json_log_handler)
+async def parse_body(request: Request):
+    data: bytes = await request.body()
+    return data
+
+
+class BasetenEndpoints(V1Endpoints):
+    def __init__(
+        self,
+        dataplane: DataPlane,
+        model_repository_extension: Optional[ModelRepositoryExtension] = None,
+    ):
+        super().__init__(dataplane, model_repository_extension)
+
+    @staticmethod
+    def check_healthy(model: ModelWrapper):
+        model.start_load()
```

Comment:
can `model.start_load()` take a long time to execute? If yes, is it ok to have that in a healthcheck method?

Reply:
The most recent push updates this a bit to use the `on_startup` method for FastAPI instead of through this check, so this line is removed. Either way, this was in a thread and wouldnt have been associated with the request.